crates/primitives/src/alloy_compat.rs

```diff
@@ -21,14 +22,20 @@ impl TryFrom<alloy_rpc_types::Block> for Block {
                     .into_iter()
                     .map(|tx| {
                         let signature = tx.signature.ok_or(ConversionError::MissingSignature)?;
+                        let recovery_id = if signature.v > U256::from(1) {
+                            signature.v - U256::from(37)
```

Comment:
this can resut in underflow/panics. there are methods on `Parity` to do this conversion. should probably eagerly convert to `Parity`